pages/dkp/konvoy/2.2/image-builder/create-azure-image/index.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: layout.pug
+navigationTitle: Create Azure Image
+title: Create a custom Azure Image
+excerpt: Learn how to build a custom Azure Image for use with Konvoy
+beta: false
+enterprise: false
+menuWeight: 60
+---
+
+This procedure describes how to use the Konvoy Image Builder to create a [Cluster API](https://cluster-api.sigs.k8s.io/) compliant Azure Image. Azure images contain configuration information and software to create a specific, pre-configured, operating environment. For example, you can create an Azure image of your current computer system settings and software. The Azure image can then be replicated and distributed, creating your computer system for other users. The Konvoy Image Builder uses variable [`overrides`][overrides] to specify base image and container images to use in your new Azure image.
+
+## Prerequisites
+
+Before you begin, you must:
+
+-   Download the latest [Konvoy Image Builder](https://github.com/mesosphere/konvoy-image-builder/releases) bundle (prefixed with `konvoy-image-bundle`) for your OS. Do not use the release prefixed with `konvoy-image-builder`.
+
+-   Create a working `Docker` setup.
+
+## Extract Bundle
+
+Extract the bundle and `cd` into the extracted `konvoy-image-bundle-$VERSION_$OS` folder. The bundled version of `konvoy-image` contains an embedded `docker` image that contains all the requirements for building.
+
+<p class="message--note"><strong>NOTE: </strong> The <code>konvoy-image</code> binary and all supporting folders are also extracted. When run, <code>konvoy-image</code> bind mounts the current working directory (<code>${PWD}</code>) into the container to be used.</p>
+
```

Comment:
Does the user needs to 'az login' and create a Service Principal to perform these actions? (i.e. should they follow the first few steps of the flow listed here: docs.d2iq.com/dkp/konvoy/2.1/choose-infrastructure/azure/quick-start-azure/#configure-azure-prerequisites).  If so:

```suggestion
## Configure Azure prerequisites

1.  Log in to Azure:

    ```bash
    az login
    ```

    ```sh
    [
      {
        "cloudName": "AzureCloud",
        "homeTenantId": "a1234567-b132-1234-1a11-1234a5678b90",
        "id": "b1234567-abcd-11a1-a0a0-1234a5678b90",
        "isDefault": true,
        "managedByTenants": [],
        "name": "Mesosphere Developer Subscription",
        "state": "Enabled",
        "tenantId": "a1234567-b132-1234-1a11-1234a5678b90",
        "user": {
          "name": "user@azuremesosphere.onmicrosoft.com",
          "type": "user"
        }
      }
    ]
    ```

1.  Create an Azure Service Principal (SP) by running the following command:

    <p class="message--note"><strong>NOTE: </strong>If an SP with the name exists, this command will rotate the password.</p>

    ```bash
    az ad sp create-for-rbac --role contributor --name "$(whoami)-konvoy"
    ```

    ```sh
    {
      "appId": "7654321a-1a23-567b-b789-0987b6543a21",
      "displayName": "azure-cli-2021-03-09-23-17-06",
      "name": "http://azure-cli-2021-03-09-23-17-06",
      "password": "Z79yVstq_E.R0R7RUUck718vEHSuyhAB0C",
      "tenant": "a1234567-b132-1234-1a11-1234a5678b90"
    }
    ```
```

Reply:
Nice, yep I'll also augment the command with `--query "{ client_id: appId, client_secret: password, tenant_id: tenant }"` so that it outputs all the need to set the env vars and the cli aparams